test/test_torch.py

```diff
@@ -5118,6 +5118,11 @@ def test_has_internal_overlap(self):
         b_expanded = b.expand(4, 3)
         self.assertEqual(torch._debug_has_internal_overlap(b_expanded), OVERLAP_YES)
 
+        # Check for zero strided, size 1 axis, in non-contiguous storage (gh-33812)
+        c = torch.randn(2, 1, 2).T
```

Comment:
Note the transpose is important, otherwise it passes the `is_contiguous()` check without testing our zero-stride.